pkg/scheduler/scheduler.go

```diff
@@ -356,7 +356,13 @@ func (sched *Scheduler) Run(ctx context.Context) {
 // recordSchedulingFailure records an event for the pod that indicates the
 // pod has failed to schedule. Also, update the pod condition and nominated node name if set.
 func (sched *Scheduler) recordSchedulingFailure(fwk framework.Framework, podInfo *framework.QueuedPodInfo, err error, reason string, nominatedNode string) {
-	sched.Error(podInfo, err)
+	// An error may be returned. For example, a Pod being deleted elsewhere or assigned a node name.
+	// In either case, we should return immediately; otherwise, the pod may be held inside
+	// scheduler queue as a phantom and cause unexpected behavior afterwards. (see #102339)
+	if err := sched.Error(podInfo, err); err != nil {
+		klog.InfoS(fmt.Sprintf("%v, abort adding it back to queue", err), "pod", klog.KObj(podInfo.Pod), "node", podInfo.Pod.Spec.NodeName)
```

Comment:
You have to adhere to this https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#remove-string-formatting-from-log-message

And you can include the error as part of the message https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#using-errors

I don't think we should include the node as a key.

Reply:
Probably we can change it to an error. Updated, PTAL.

Reply:
InfoS should be fine, but that's a nit.